Fix knight crab NM status

TracentEden opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->

**_I affirm:_**

- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/CODE_OF_CONDUCT.md).
- [ ] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## Please enter a player-facing description
Knight crabs are now considered notorious monsters and will no longer despawn unless they are killed. (Tracent)

## What does this pull request do? (Please be technical)
This PR adds the notorious monster tag to knight crabs to prevent them from despawning unless killed.

## Steps to test these changes
Zone knight crabs and they should path back to spawn points.

## Special Deployment Considerations

<!-- Include any steps that need to be taken when deploying to the live environment. -->
<!-- Example: Need to run one_time_sql_conversion.sql -->

CriticalXI commented:
I took a look at a capture for King Arthro and can confirm that the Knight Crabs are not NMs. We may likely need to create a new mobmob to not despawn on deaggro.

TracentEden commented:
> I took a look at a capture for King Arthro and can confirm that the Knight Crabs are not NMs. We may likely need to create a new mobmob to not despawn on deaggro.

Yeah, luckily there already exists such a mod, I updated the PR to add it knight crabs and removed the part that made them NMs.
